htlcswitch/link.go

```diff
@@ -2083,9 +2097,36 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			}
 		}
 
+		// Lookup the original htlc, if it was part of a blinded route
+		// switch out our error.
```

Comment:
another understanding check: if we receive an `UpdateFailHTLC` from our peer when we are also in the blinded route, then they are breaking the rules right? Cause they _should_ only send us `UpdateFailMalformed`... right?

Reply:
Yeah 👿